ui/components/NodeInputSubmit.tsx

```diff
@@ -42,9 +42,21 @@ export const NodeInputSubmit: FC<NodeInputProps> = ({
   const showTotpLink =
     (node.meta.label as unknown as { hasTotpLink: boolean })?.hasTotpLink ??
     false;
+  const showBackLink = node.meta?.label?.text === "Reset password";
 
   return (
     <>
+      {showBackLink && (
+        <Button
+          tabIndex={3}
+          type="button"
+          onClick={() => {
+            void window.history.back();
```

Comment:
IMHO back buttons are a pain both from a dev and a UX perspective.

question: what happens if I access this page directly? My guess is that I will be redirected back to wherever I was, @lukasSerelis is that we want or do we always want to be redirected to the login page? What happens if I am logged in and get redirected to the login page? IMO if we want to add back buttons, we should hard code in the UI the succession of pages in the flow and redirect the user there based on these hardcoded values, not based on the browser's history stack.

Reply:
The problem with hardcoding is, that we lose the challenge from the app. So when a user came from grafana, they won't log in to grafana any more. Using history was the best of all worlds I could think of. Technically, this is always the correct behaviour.

Reply:
Oh, that's right. I didn't think of this.

Then I am not sold on whether we need this back button, but I will let it be :P

Reply:
Happy to remove it, as that simplifies the code. @lukasSerelis wdyt?

Reply:
Even if this is just a call for history back I would like to keep the back button. Login flows are notoriously finicky with going back (similar to checkout flows), so being explicit with back actions here is helpful. Otherwise, it's not clear at all what are the options if I am not sure I want to reset anymore.